src/widget/tool/profileimporter.cpp

```diff
@@ -0,0 +1,65 @@
+#include "profileimporter.h"
+#include <QString>
+#include <QFile>
+#include <QDir>
+#include <QFileDialog>
+#include <QFileInfo>
+#include "src/persistence/settings.h"
+#include "src/core/core.h"
+#include "src/widget/gui.h"
+#include <QDebug>
+#include <QMessageBox>
+
+ProfileImporter::ProfileImporter(QWidget *parent) : QWidget(parent)
+{
+
+}
+
+bool ProfileImporter::importProfile()
+{
+    QString path = QFileDialog::getOpenFileName(this,
```

Comment:
I learnt this "trick" when I was starting with Qt couple of years ago and it worked so far ;)

Your solution is better so I'm reverting those changes, I will try it on my private project and if worked well it will be ported to qtox.